java/java.source.base/src/org/netbeans/modules/java/source/usages/BuildArtifactMapperImpl.java

```diff
@@ -853,8 +870,9 @@ private Boolean performUpdate(@NonNull final Context ctx) throws IOException {
                 File target = resolveFile(targetFolder, relPath);                        
 
                 try {
-                    copyFile(updatedFile, target);
-                    updatedFiles.add(target);
+                    if (copyFile(updatedFile, target, new URL(ctx.getSourceRoot(), relPath))) {
+                        updatedFiles.add(target);
```

Comment:
What's the rule for which files need to be added to updatedFiles? Who is listening for changes here? If the reason that the file was not copied is that it changed externally, is that something a listener somewhere should be notified of?

Reply:
I do not think so. The idea here is that a build created/updated, say, `target/classes/pkg/Clazz.class` so it now has a newer timestamp than `Clazz.sig` and we should not clobber it with an older and possibly not-quite-correct version. There is nothing to listen to. If the next thing that happens is another external build, the timestamp will be even newer, and so the next CoS run will again trust the `Clazz.class`. If on the other hand the user edits & saves `Clazz.java`, causing `Clazz.sig` to be updated, and next does a CoS run, then we will overwrite `Clazz.class` with the contents of `Clazz.sig`.

Reply:
Makes sense.